release/models/hashing/openconfig-hashing.yang

```diff
@@ -0,0 +1,332 @@
+            compute the hash.";
+        leaf ipv4-none {
+            type boolean;
+            description "Do not use any IPv4 fields in the
+                calculation of the hash.";
+        }
+        leaf ipv4-src-address {
+            type boolean;
+            description "Use only the IPv4 source address in the
+                calculation of the hash.";
+        }
+        leaf ipv4-dst-address {
+            type boolean;
+            description "Use only the IPv4 destination address in the
+                calculation of the hash.";
+        }
+        leaf ipv4-both {
+            type boolean;
+            description "Use both the IPv4 source and
+                destination addresses in the calculation of the
+                hash.";
+        }
+        leaf ipv4-target-define {
+            type boolean;
+            description "Use the default fields that are used by
+                the device for calculating the hash.";
+        }
+        leaf ipv4-ingress-subinterface {
+            type boolean;
+            description "Use the ingress subinterface identifired
+                in the calculation of the hash.";
+        }
+    }
+
+    grouping ipv4-fields-top {
+        description
+            "Top-level container of hashing input for ipv4";
+        container config {
+            description
+                "Configuration data of hashing input for ipv6";
+            uses ipv4-fields-config;
+        }
+        container state {
+            description
+                "Operational data of hashing input for ipv6";
+            config false;
+            uses ipv4-fields-config;
+        }
+    }
+
+    grouping ipv6-fields-config {
+        description "The IPv6 fields that should be used to
+            compute the hash.";
+        leaf ipv6-none {
+            type boolean;
+            description "Do not use any IPv6 fields in the
+                calculation of the hash.";
+        }
+        leaf ipv6-src-address {
+            type boolean;
+            description "Use only the IPv6 source address in the
+                calculation of the hash.";
+        }
+        leaf ipv6-dst-address {
+            type boolean;
+            description "Use only the IPv6 destination address in the
+                calculation of the hash.";
+        }
+        leaf ipv6-both {
+            type boolean;
+            description "Use both the IPv6 source and
+                destination addresses in the calculation of the
+                hash.";
+        }
+        leaf ipv6-target-define {
+            type boolean;
+            description "Use the default fields that are used by
+                the device for calculating the hash.";
+        }
+        leaf ipv6-flow-label {
+            type boolean;
+            description "Use the flow label in the IPv6 header
+                to calculate the hash.";
+        }
+        leaf ipv6-ingress-subinterface {
```

Comment:
I think we can leave this off and instead define a grouping with just one boolean for 'ingress-interface' (vs. subinterface).

Reply:
```
    grouping ingress-interface-top {
        leaf ingress-interface {
            type boolean;
            description "Use the ingress subinterface identifired in the
                calculation of the hash.";
        }
    }

grouping ipv6-fields-config {
    ...
    uses ingress-interface-top;
}
grouping ipv4-fields-config {
    ...
    uses ingress-interface-top;
}
```

Reply:
Yes, that is what I'm thinking

Reply:
Ah, ingress-interface should be instantiated at:  `/system/hashing-policies/hashing-policy/config/ingress-interface`